frontend/src/components/BackgroundEffects/BackgroundGallery/BackgroundGallery.spec.tsx

```diff
@@ -0,0 +1,34 @@
+import { describe, expect, it, vi } from 'vitest';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import BackgroundGallery, { backgrounds } from './BackgroundGallery';
+
+describe('BackgroundGallery', () => {
+  const backgroundsFiles = backgrounds.map((bg) => bg.file);
+
+  it('renders all background images as selectable options', () => {
+    render(<BackgroundGallery backgroundSelected="" setBackgroundSelected={() => {}} />);
+    const imgs = screen.getAllByRole('img');
+    backgroundsFiles.forEach((file) => {
+      expect(imgs.some((img) => (img as HTMLImageElement).src.includes(file))).toBe(true);
+    });
+    const options = backgrounds.map((bg) => screen.getByTestId(`background-${bg.id}`));
+    expect(options).toHaveLength(backgroundsFiles.length);
+  });
+
+  it('calls setBackgroundSelected when an option is clicked', async () => {
```

Comment:
nit: the test name is too direct, something like `sets the selected background when foo` would be better